features/dd-sdk-android-ndk-tracer/src/main/kotlin/com/datadog/android/ndk/tracer/MetricsCollector.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.ndk.tracer
+
+import java.io.File
+import java.util.LinkedList
+import kotlin.system.measureNanoTime
+
+internal class MetricsCollector(private val methodName: String, metricsCollectorDirectory: File? = null) {
+
+    private var durationInMs: Double = 0.0
+    private var count: Long = 0
+    private val metricsHistory:LinkedList<Double> = LinkedList()
+    private val metricsCollectorFile: File? = metricsCollectorDirectory?.let {
+        File(it, "metrics_collector_$methodName.txt")
+    }
+
+    internal fun measureMethodDuration(block: () -> Unit) {
+
+        val duration = measureNanoTime(block).toMilliseconds()
+        addDuration(duration)
+    }
+
+    private fun Long.toMilliseconds(): Double {
+        return this / 1_000_000.0
```

Comment:
We definitely should move such extension somewhere in common code. I saw such transformation in every second feature 😄